app/scripts/metamask-controller.js

```diff
@@ -5713,10 +5789,14 @@ export default class MetamaskController extends EventEmitter {
         },
       );
       if (isHdKeyring) {
-        await this._addAccountsWithBalance(
-          metadata.id,
-          shouldImportSolanaAccount,
-        );
+        if (this.isMultichainAccountsState2Enabled()) {
```

Comment:
Do we need to also check for `shouldImportSolanaAccount` here too?

Reply:
Nope, the code path intends on Solana being discovered here so no need to add the check. `_importAccountsWithBalance` is only ever called when onboarding is completed.